linker/descriptors_test.go

```diff
@@ -63,6 +66,39 @@ func TestFields(t *testing.T) {
 	}
 }
 
+func TestUnescape(t *testing.T) {
+	t.Parallel()
+	fileProto := &descriptorpb.FileDescriptorProto{
+		Name:   proto.String("foo.proto"),
+		Syntax: proto.String("proto2"),
+		MessageType: []*descriptorpb.DescriptorProto{
+			{
+				Name: proto.String("Foo"),
+				Field: []*descriptorpb.FieldDescriptorProto{
+					{
+						Name:         proto.String("escaped_bytes"),
+						DefaultValue: proto.String(`\p\0\001\02\ab\b\f\n\r\t\v\\\'\"\?\xfe\Xab\u2192\U0001F389`),
```

Comment:
Tests the following:
* Values that don't need escaping.
* Octal 1/2/3 length.
* All escape characters.
* Upper/lowercase hex.
* Short/long unicode.

Reply:
Per https://github.com/bufbuild/protobuf.com/pull/6, I think we should also have cases for only one hex digit for `\x` and `\X`.